### PR TITLE
Nfc: fix incorrect type castings. Global: fix printf usage, types casting, overall cleanup. Drivers: incorrect array index in cc1101 driver.

### PR DESCRIPTION
**Pull request by tonyfreeman**
# What's new

- Nfc: fix incorrect type castings.
- Global: fix printf usage, types casting, overall cleanup.
- Drivers: incorrect array index in cc1101 driver.

# Verification 

- Compile and upload
- Try to compile with Clang

# Checklist (do not modify)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: applications/nfc/scenes/nfc_scene_read_card_success.c [hunk 1]
```diff
@@ -22,7 +22,8 @@ void nfc_scene_read_card_success_on_enter(void* context) {
     DialogEx* dialog_ex = nfc->dialog_ex;
     dialog_ex_set_left_button_text(dialog_ex, "Retry");
     dialog_ex_set_right_button_text(dialog_ex, "More");
-    dialog_ex_set_header(dialog_ex, nfc_get_dev_type(data->device), 36, 8, AlignLeft, AlignCenter);
+    dialog_ex_set_header(
+        dialog_ex, nfc_get_dev_type((rfalNfcDevType)data->device), 36, 8, AlignLeft, AlignCenter);
```
Comment (skotopes) [inline applications/nfc/scenes/nfc_scene_read_card_success.c]: This place looks incorrect. @gornekich

File: applications/nfc/scenes/nfc_scene_save_name.c [hunk 2]
```diff
@@ -32,12 +32,12 @@ const void nfc_scene_save_name_on_enter(void* context) {
...
 }
 
-const bool nfc_scene_save_name_on_event(void* context, SceneManagerEvent event) {
+bool nfc_scene_save_name_on_event(void* context, SceneManagerEvent event) {
     Nfc* nfc = (Nfc*)context;
 
     if(event.type == SceneManagerEventTypeCustom) {
         if(event.event == SCENE_SAVE_NAME_CUSTOM_EVENT) {
-            if(nfc->dev.dev_name) {
+            if(strcmp(nfc->dev.dev_name, "")) {
```
Comment (skotopes) [inline applications/nfc/scenes/nfc_scene_save_name.c:40]: @gornekich what kind of logic expected here?

File: applications/storage/storages/storage-ext.c [hunk 3]
```diff
@@ -99,7 +99,8 @@ FS_Error sd_unmount_card(StorageData* storage) {
     SDError error;
 
     storage_data_lock(storage);
-    error = storage->status = StorageStatusNotReady;
+    storage->status = StorageStatusNotReady;
```
Comment (skotopes) [inline applications/storage/storages/storage-ext.c:102]: Looks legit.

@DrZlo13 ?

File: firmware/targets/f6/furi-hal/furi-hal-bt.c [hunk 4]
```diff
@@ -61,7 +61,7 @@ bool furi_hal_bt_is_active() {
 }
 
 bool furi_hal_bt_wait_startup() {
-    uint8_t counter = 0;
+    uint16_t counter = 0;
```
Comment (skotopes) [inline firmware/targets/f6/furi-hal/furi-hal-bt.c:64]: F7 have same issue.

File: applications/storage/storages/storage-ext.c (hunk 3 above)
Comment (DrZlo13) [reply applications/storage/storages/storage-ext.c:102]: Yeah.

File: applications/nfc/scenes/nfc_scene_save_name.c (hunk 2 above)
Comment (gornekich) [reply applications/nfc/scenes/nfc_scene_save_name.c:40]: Fix is correct

File: applications/nfc/scenes/nfc_scene_read_card_success.c (hunk 1 above)
Comment (gornekich) [reply applications/nfc/scenes/nfc_scene_read_card_success.c]: Fixed

**Review by gornekich — DISMISSED** [review]

**Review by skotopes — APPROVED** [review]
Nice work.

**Pull request merged**